src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -187,9 +185,14 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
 
         if ('' === $url) {
             $url = '/';
-        } elseif (preg_match($this->urlEncodingSkipRegexp, $url)) {
-            // the context base URL is already encoded (see Symfony\Component\HttpFoundation\Request)
+        } elseif (!empty($this->decodedChars)) {
+            @trigger_error('The class variable '.__CLASS__.'::$decodedChars is deprecated since version 2.7 and will be removed in 3.1.', E_USER_DEPRECATED);
```

Comment:
@arjenm deprecations cannot be made in existing minor releases. A deprecation implies that there's a new way of doing something, also known as a feature. In order to add a feature, it has to be in the newest minor release (3.1, 3.2 etc). This PR is requested against 2.7, which doesn't accept new features anymore (same goes for any other lower than 3.1 at the moment as 3.1 is the next).

So what I think you should do, is reopen this PR against the master if you want it as a feature. This means you will have to update your message to something like "deprecated in 3.1, removed in 4.0". Currently 2.7 and 2.8 only accept bug fixes (will be explained a bit better in the docs: symfony/symfony-docs#6412).